### PR TITLE
fix(jira-server): Fetch project priorities using safe API

### LINKED ISSUE #58906: "Something went wrong" trying to create Jira issue from Sentry issue
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. Navigate to https://benchling.sentry.io/issues/4577254609/ (need permissions, naturally)
2. Click the plus icon by "Jira Server Issue".
3. See the popup to choose between creating a new issue or linking to an old one _briefly_ (can't do anything)
4. See the popup change to this: 
Screenshot 2023-10-26 at 5 48 21 PM

### Expected Result

Being able to create a new Jira or link to an existing one

### Actual Result

see screenshot in "steps to reproduce"

### Product Area

Issues

### Link

https://benchling.sentry.io/issues/4577254609/

### DSN

_No response_

### Version

_No response_

### PR DESCRIPTION
**Pull request by schew2381**
‼️ Reverted b/c this runs into the same permission issue as https://github.com/getsentry/sentry/pull/58749 despite the Jira Server API docs not stating I needed global/admin permission. This is the relevant [Sentry Issue](https://sentry.sentry.io/issues/4593933722/events/6886ed5320694ad8a4925816ba9c254a/?project=1&referrer=issue-list&statsPeriod=10m)

Fixes https://github.com/getsentry/sentry/issues/58906

I changed the way we fetch Jira Server priorities to scope them per project in https://github.com/getsentry/sentry/pull/58749. However I used [getAssignedPriorityScheme](https://docs.atlassian.com/software/jira/docs/api/REST/9.11.0/#api/2/project/{projectKeyOrId}/priorityscheme-getAssignedPriorityScheme) which requires project/global admin permission.

This fix switches to using [getPrioritySchemes](https://docs.atlassian.com/software/jira/docs/api/REST/9.11.0/#api/2/priorityschemes-getPrioritySchemes) where we first fetch all priority schemes, then match the response with the project key to find the priority IDs.

### Auto-Ticket Creation

https://github.com/getsentry/sentry/assets/67301797/4dab6974-c8b9-4788-a683-af6c53f5e589

### Manual Ticket Creation

https://github.com/getsentry/sentry/assets/67301797/e31a7c6c-005e-4ed6-b5a4-099411c4e149

### CODE REVIEW

File: src/sentry/integrations/jira_server/client.py [hunk 1]
```diff
@@ -36,8 +36,8 @@ class JiraServerClient(IntegrationProxyClient):
     ISSUE_TYPES_URL = "/rest/api/2/issue/createmeta/%s/issuetypes"
     # https://docs.atlassian.com/software/jira/docs/api/REST/9.11.0/#api/2/priority-getPriorities
     PRIORITIES_URL = "/rest/api/2/priority"
-    # https://docs.atlassian.com/software/jira/docs/api/REST/9.11.0/#api/2/project/{projectKeyOrId}/priorityscheme
-    PRIORITY_SCHEME_URL = "/rest/api/2/project/%s/priorityscheme"
+    # https://docs.atlassian.com/software/jira/docs/api/REST/9.11.0/#api/2/priorityschemes-getPrioritySchemes
+    PRIORITY_SCHEMES_URL = "/rest/api/2/priorityschemes?expand=schemes.projectKeys&maxResults=200"
```
Comment (schew2381) [inline src/sentry/integrations/jira_server/client.py:40]: The default maximum results is 100, but I changed it to 200 to be safe. I'd be very surprised if anybody has more priority schemes than that

File: src/sentry/integrations/jira_server/client.py [hunk 2]
```diff
@@ -147,15 +147,27 @@ def get_project_key_for_id(self, project_id) -> str:
...
         """
-        Fetches all priorities associated with the project. Jira Server does not provide a
-        single API to fetch priorities for a project, so we instead fetch the project's
-        priority scheme which contains the relevant priority ids (but not human-readable names).
-        We then fetch the full list of priorities and find the overlap to get both ids and names.
+        Fetches all priorities associated with the project. Jira Server does not provide a single API to
+        fetch priorities for a project, so we instead fetch all priority schemes and find the one that
+        applies to the project. Priority schemes only contain priority ids (not human-readable), so we also
+        need to fetch the full list of priorities and find the overlap to get both ids and names.
+        XXX: Note that we can't use getAssignedPriorityScheme b/c it requires either global or proj administrator.
```
Comment (sentaur-athena) [inline src/sentry/integrations/jira_server/client.py:156]: nit: 
```suggestion
        XXX: Note that we can't use getAssignedPriorityScheme b/c it requires either global or proj administrator permission.
```

File: tests/sentry/integrations/jira_server/test_integration.py [hunk 3]
```diff
@@ -363,16 +363,36 @@ def test_get_create_issue_config_with_project_having_subset_priorities(self):
 
         responses.add(
             responses.GET,
-            f"https://jira.example.org/rest/api/2/project/{DEFAULT_PROJECT_ID}/priorityscheme",
+            "https://jira.example.org/rest/api/2/priorityschemes?expand=schemes.projectKeys&maxResults=200",
```
Comment (sentaur-athena) [inline tests/sentry/integrations/jira_server/test_integration.py:366]: nit: make from PRIORITY_SCHEMES_URL

File: tests/sentry/integrations/jira_server/test_integration.py [hunk 4]
```diff
@@ -428,16 +440,108 @@ def get_priorities_callthrough_client(installation: JiraServerIntegration):
             priority_field = [field for field in fields if field["name"] == "priority"][0]
             assert priority_field == {
                 "default": "",
+                # Choices should exclude the low priority
```
Comment (sentaur-athena) [inline tests/sentry/integrations/jira_server/test_integration.py:443]: why?

**Review by sentaur-athena — APPROVED** [review]

File: tests/sentry/integrations/jira_server/test_integration.py (hunk 4 above)
Comment (schew2381) [reply tests/sentry/integrations/jira_server/test_integration.py:443]: The priority scheme we match to the current project only has the `High` priority, see [here](https://github.com/getsentry/sentry/pull/59197/files#diff-63c86a8a3b908c4e19d76f3f9daf5141b2b6b65b93baa6ee81db838566d2014aR388-R390)

**Pull request merged**